### ISSUE TITLE
SepformerSeparation audios

### ISSUE THREAD

**Issue by sharathc10** [speechbrain/speechbrain#790]
I was successfully able to use the SepformerSeparation class for a wav file with two speakers(S1,S2).  The output files 'source1hat.wav' has S2 slightly muted and 'source2hat.wav' has S1 slightly muted which is great.  However, I need to transcribe these files and the asr_model.transcribe_file is picking up the muted speech.  Is there a way to remove the muted speech altogether?  Thanks for your help. 

Also, this doesn't work for longer audios.  Should I be running it on Colab?

**Comment by mravanelli** [comment]
Hi @sharathc10, here @ycemsubakan  can help. I'm not sure to understand the problem. The speech recognizer is going to transcribe the file that you give it in input. What do you mean with "muted"?

**Comment by ycemsubakan** [comment]
Hi @sharathc10 , what model are you using? We have whamr, whamr16k, and WSJ02Mix models.

**Comment by sharathc10** [comment]
Please see my code below - it is wsj02mix.  By muted I mean that the sound is not audible or is softened, which is fine but the transcription is picking up the muted sounds.

`from speechbrain.pretrained import SepformerSeparation as separator
import torchaudio
model =separator.from_hparams(source="speechbrain/sepformer-wsj02mix",
savedir='pretrained_models/sepformer-wsj02mix')
# for custom file, change path
est_sources =model.separate_file('chatbot.wav') 
torchaudio.save("source1hat3.wav", est_sources[:, :, 
0].detach().cpu(), 8000)
torchaudio.save("source2hat3.wav", est_sources[:, :, 
1].detach().cpu(), 8000)`

**Comment by ycemsubakan** [comment]
@sharathc10 , It seems like you are talking about a fundamental research question. 

This being said, I also suggest you try sepformer-whamr, and sepformer-wham.

**Comment by sharathc10** [comment]
Thanks for your recommendations.  I tried the models you suggested @ycemsubakan.  It seems to be doing the same thing i.e. mute one of the speakers.  I just need to figure out how to remove the muted sections.  Do you have any suggestions?

**Comment by ycemsubakan** [comment]
Can you share the separated examples? You can use https://www.sndup.net/ to share.

**Comment by sharathc10** [comment]
The parent file is the original recording and the next two files are the outputs from the following model - speechbrain/sepformer-whamr16k.  You will hear the muted parts in the separated files.  Is there a way to just remove them?

Parent file
https://sndup.net/3h7q

Separated files
https://sndup.net/5rgf
https://sndup.net/5ysc

Here is my code:

`
from speechbrain.pretrained import SepformerSeparation as separator
import torchaudio
model =separator.from_hparams(source="speechbrain/sepformer-wsj02mix",
savedir='pretrained_models/sepformer-wsj02mix')
# for custom file, change path
est_sources =model.separate_file('chatbotc1_16000.wav') 
torchaudio.save("source1hat.wav", est_sources[:, :, 
0].detach().cpu(), 8000)
torchaudio.save("source2hat.wav", est_sources[:, :, 
1].detach().cpu(), 8000)
`

**Comment by ycemsubakan** [comment]
The first thing I suspect is the sampling rate. Note that this model is trained to separate signals with 8kHz sampling frequency. 
Your file's name has 16000, are you sure it is sampled at 8 kHz? 

If not, you can try our 16kHz model: https://huggingface.co/speechbrain/sepformer-whamr16k
Or of course you can try downsampling the input signal to 8kHz.

**Comment by ycemsubakan** [comment]
I tried it with 16k model as well, it works slightly better. But it's not perfect. This is basically the performance we observe. One thing you can do as postprocessing step would be to threshold based on energy. That can help you eliminate some of the artifacts, and remaining undesired speech from the other source. I am closing the issue as it is not related to a bug.

**Comment by sharathc10** [comment]
Thanks for your help.
